va/http_test.go

```diff
@@ -774,9 +823,17 @@ func TestFetchHTTP(t *testing.T) {
 			Host: "example.com",
 			Path: "/loop",
 			ExpectedProblem: probs.ConnectionFailure(fmt.Sprintf(
-				"Fetching http://example.com:%d/loop: Too many redirects", httpPort)),
+				"Fetching http://example.com:%d/loop: Redirect loop detected", httpPort)),
 			ExpectedRecords: expectedLoopRecords,
 		},
+		{
+			Name: "Too many redirects",
+			Host: "example.com",
+			Path: "/max-redirect",
```

Comment:
I think that `/max-redirect` here and `/loop` on line 824 above should be changed to `<httpPort>:/max-redirect/0` and `<httpPort>:/loop` respectively. This would allow all of the `if i != 0` special-casing on lines 748 and 768 to be removed (because the first record would already have the port in it), would allow `maxRedirect+1` on line 765 to become just `maxRedirect` (because we're already starting with 0), and would allow the one-off HandleFunc on line 520 to be removed entirely. Overall, this change would make two lines (here in this block of test configs) _slightly_ more complex, but would greatly simplify all the setup code above, so I think it's worth it.

Reply:
After @aarongable and I looked into this more it would be a more complex change than first thought and not necessary right now. The initial validation having a port added in the test creates a problem with the preresolvedDialer, either in http.go or http_test.go that is not worth messing with.